streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -134,10 +136,7 @@ public void register(final StateStore store,
 
         final TopicPartition storePartition = new TopicPartition(topic, getPartition(topic));
 
-        final StateStore stateStore =
-            store instanceof WrappedStateStore ? ((WrappedStateStore) store).inner() : store;
-        final RecordConverter recordConverter =
-            stateStore instanceof TimestampedBytesStore ? RecordConverter.converter() : record -> record;
+        final RecordConverter recordConverter = isTimestamped(store) ? rawValueToTimestampedValue() : identity();
```

Comment:
This is the same as `GlobalStateManagerImpl.java` -- maybe add method `AbstractStateManager#recordConverter(store)`  to unify code?